### DISCUSSION TITLE
Mastodon instance gets connection timeouts when trying to send messages to ntfy.sh

### DISCUSSION THREAD

**Discussion by pabloa** [mastodon/mastodon#22677] [Q&A]
When my mastodon instance tries to send messages to ntfy.sh, it fails by timeout.

I tried to send a message using curl from my instance, and I could not do it from my host. 
I can send messages to nifty.sh without issues from anywhere else.

I don't know how to proceed. 
Should I configure something else so messages can be delivered to ntfy.sh broker? Should I configure another broker? If I launch my ntfy instance, will the mastodon clients receive them?

Could you let me know how you solved this issue? I googled it, and I could not find a solution. 

I would greatly appreciate any advice you can give me. 

<html><head></head><body>

  | just now | 4 | push | Web::PushNotificationWorker | 6479, 2060856 | HTTP::TimeoutError: Connect timed out after 5 seconds on https://ntfy.sh/upwCHsNs9tCXxg?up=1
-- | -- | -- | -- | -- | -- | --

</body></html>[just now](https://mymastodon.com/sidekiq/retries/1671770994.6359947-3e0fba80b9d036a5985101f8) 	4 	[push](https://mymastodon.com/sidekiq/queues/push) 	Web::PushNotificationWorker 	
6479, 2060856
	
HTTP::TimeoutError: Connect timed out after 5 seconds on https://ntfy.sh/upwCHsNs9tCXxg?up=1

**Comment by rbairwell** [comment]
>  tried to send a message using curl from my instance, and I could not do it from my host

If you can't send the message from curl, that rules it out being a Mastodon software issue.

It sounds like either a firewall, routing or possibly SSL certificate issue. If you run `traceroute ntfy.sh` from your Sidekiq instance what do you get? Same with running `curl -vvv https://ntfy.sh`.

**Comment by thardie** [comment]
This happens when you push too much traffic to ntfy.sh. They will start to block you. They first respond with 429 (which mastodon doesn't back off from), and keeps trying, so they eventually firewall your IP.

**Reply by inchw0rm** [reply]
Hey that's pretty helpful, but how do I prevent this from happening?

**Comment by ShadowJonathan** [comment]
https://github.com/binwiederhier/ntfy/issues/664, https://github.com/binwiederhier/ntfy/pull/842 and https://github.com/binwiederhier/ntfy/pull/979 are relevant here
